### PR TITLE
Convert index2 to reverse complementary sequence for HiSeqX samplesheet

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: scripts/samplesheet_generator.py [hunk 1]
```diff
@@ -82,7 +83,7 @@ def gen_X_lane_data(pro):
             idxs = find_barcode(sample, pro)
             sp_obj['idx1'] = idxs[0].replace(',','')
             try:
-                sp_obj['idx2'] = idxs[1].replace(',','')
+                sp_obj['idx2'] = str(Seq(idxs[1].replace(',','')).reverse_complement())
```
Comment (ewels) [inline scripts/samplesheet_generator.py]: I think that you can probably do this fairly easily in pure python without introducing a new package dependency...

```python
compl = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'} 
sp_obj['idx2'] = ''.join( reversed( [compl.get(b,b) for b in idxs[1].replace(',','') ] ) )
```

(untested)

**Comment by chuan-wang** [comment]
Thanks Phil! I didn't think about this point. I've tested your method and it worked as expected.

**Comment by ewels** [comment]
Great! Just thought - do we ever have lower case letters in sample sheets? My solution will probably be case specific. (in such cases it would just return the reverse, rather than the reverse compliment).

Easy to add a safety fix there - just add `.upper()` to the string first.. `idxs[1].replace(',','').upper()`

**Comment by chuan-wang** [comment]
I remember that LIMS converts indexes to upper case. But it's nice to add this solution for making sure that things will work :)

**Pull request merged**